<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OptionsWindow</class>
 <widget class="QDialog" name="OptionsWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>493</width>
    <height>353</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Options</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QListWidget" name="option">
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>16777215</height>
        </size>
       </property>
       <item>
        <property name="text">
         <string>Joystick</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Servos</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Serial Port</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QStackedWidget" name="stack">
       <property name="currentIndex">
        <number>2</number>
       </property>
       <widget class="QWidget" name="joystick">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Devices connected: </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="joyN">
             <property name="text">
              <string>#</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Selected</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="joySel"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>X movement</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="joyMX"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_14">
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Y movement</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="joyMY"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Z movement</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="joyMZ"/>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>130</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_18">
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="servoRefresh">
             <property name="text">
              <string>Refresh</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="servos">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Arm 1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="servo0"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Arm 2</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="servo1"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Arm 3</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="servo2"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>Speed (%)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox">
             <property name="minimum">
              <number>10</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="singleStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>60</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>61</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Clamp</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="servo3"/>
           </item>
           <item>
            <widget class="QCheckBox" name="servo3A">
             <property name="text">
              <string>Activated</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Rotation</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="servo4"/>
           </item>
           <item>
            <widget class="QCheckBox" name="servo4A">
             <property name="text">
              <string>Activated</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>60</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="serialPort">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Ports connected:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="portN">
             <property name="text">
              <string>#</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Servo Port</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="portS"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Baud Rate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="baudRS">
             <property name="prefix">
              <string/>
             </property>
             <property name="minimum">
              <number>300</number>
             </property>
             <property name="maximum">
              <number>2000000</number>
             </property>
             <property name="singleStep">
              <number>100</number>
             </property>
             <property name="value">
              <number>1000000</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Clamp Port</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_17">
           <item>
            <widget class="QLabel" name="label_18">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="portC"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>Baud Rate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="baudRC">
             <property name="prefix">
              <string/>
             </property>
             <property name="minimum">
              <number>300</number>
             </property>
             <property name="maximum">
              <number>2000000</number>
             </property>
             <property name="singleStep">
              <number>100</number>
             </property>
             <property name="value">
              <number>9600</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>186</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OptionsWindow</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OptionsWindow</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>option</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>stack</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>233</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
